testing/dart/platform_dispatcher_test.dart

```diff
@@ -39,4 +44,60 @@ void main() {
       return ViewConfiguration(view: view)..copyWith(view: view, window: view);
     });
   });
+
+  test("PlatformDispatcher and PlatformConfiguration have matching getters", () {
+      final List<Pattern> exclusionCriteria = <Pattern>[
+        RegExp(r'^_.*'), // exclude any private members
+        RegExp(r'^on[A-Z].*'), // exclude any callbacks
+        RegExp(r'locale$'), // locale is a convenience getter for interacting with `locales` which _is_ backed by the config
+        'instance', // exclude the static instance of PlatformDispatcher
+        'configuration', // the configuration should not reference itself
+        'views', // views are backed by their own config and don't need to be referenced in the platform config
+        'initialLifecycleState', // default route is stored/retrieved from the platform, not the engine
+        'frameData', // framed data updates too often and would kick off too many config changes
+      ];
+
+      final String flutterDir = Platform.environment['FLUTTER_DIR']!;
+      final List<ClassDeclaration> classes = parseFile(
+        path: '$flutterDir/lib/ui/platform_dispatcher.dart',
+        featureSet: FeatureSet.latestLanguageVersion(),
+        throwIfDiagnostics: false,
+      ).unit.declarations.whereType<ClassDeclaration>().toList();
+      final ClassDeclaration dispatcherClass = classes.singleWhere((ClassDeclaration c) => c.name.toString() == (PlatformDispatcher).toString());
+      final ClassDeclaration configurationClass = classes.singleWhere((ClassDeclaration c) => c.name.toString() == (PlatformConfiguration).toString());
+
+      Iterable<String> getNameOfClassMember(ClassMember member) {
+        if (member is MethodDeclaration) {
+          return <String>[member.name.toString()];
+        }else if (member is FieldDeclaration) {
+          return member.fields.variables.map((VariableDeclaration variable) => variable.name.toString());
+        } else {
+          return [];
+        }
+      }
+
+      final Set<String> dispatcherMembers = dispatcherClass.members
+        .where((ClassMember member) =>
+          member is FieldDeclaration ||
+          member is MethodDeclaration && member.isGetter)
+        .expand<String>(getNameOfClassMember)
+        .where((String name) =>
+          exclusionCriteria.every((criteria) =>
+            criteria.allMatches(name).isEmpty))
+        .toSet();
+      final Set<String> configurationMembers = configurationClass.members
+        .where((ClassMember member) =>
+          member is FieldDeclaration ||
+          member is MethodDeclaration && member.isGetter)
+        .expand(getNameOfClassMember)
+        .toSet();
+
+      final Set<String> missingMembers = configurationMembers.difference(dispatcherMembers);
+      final Set<String> extraMembers = dispatcherMembers.difference(configurationMembers);
+      expect(missingMembers.isEmpty && extraMembers.isEmpty, isTrue,
```

Comment:
This test looks a little bit funky to me, isn't the "exposes the same getters" an OOP concern (like a "PlatformConfigurationProvider" mixin or similar) rather than something that needs to be asserted in a test?

The compiler is good at flagging these things :P

Reply:
I'd normally agree with you, but the issue that let to this work in the first place was that someone added to a class without adding to another class. I don't think that adding an _additional_ place that things need to be added/implemented is going to resolve that, where this test does give a warning in the same situation of "someone added to PlatformDispatcher without following up elsewhere".

That said, the test requires `package:analyzer`, and I'm not sure I'll be able to swing pulling that dependency into the tests, so it may be a moot point.

Reply:
I'm not trying to block this at all. I'm just saying that needing a test like this is a smell of something that needs to be solved elsewhere.

It might be an OOP/design issue (lacking an interface, or maybe exposing the `config` instance from the dispatcher so people use getters from it, rather than having this 1:1 mapping), or maybe having 1.3k lines in a file makes it harder for people to reason about *where* and how to add things (and the code reviewer to pick up those differences).

Reply:
Fair points all. I've talked with @goderbauer a bit about this, and we were actually floating the idea of _removing_ the config part of the equation. We weren't able to track down the original reasoning for having the config in the first place though, so we weren't able to tell how impactful actually removing it would be.

I personally want to leave the config for now, possibly file an issue or something to re-evaluate whether `PlatformDispatcher` and `FlutterView` actually _need_ the configs, or whether they can be simplified.